### PR TITLE
feat: add .mailmap-exclude

### PR DESCRIPTION
**Pull request by rohaquinlop**
- Add .mailmap-exclude file to exclude authors from the mailmap
- Update README.md
- Update the linter to verify the existence of the .mailmap-exclude file
- Update the linter to read the .mailmap-exclude
- Update the linter to exclude authors from the .mailmap

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,100 +1,121 @@
...
+            echo '  [ERROR] Please add a .mailmap file in the root of the repo'
+            exit 1
+          }
+
+    echo '[INFO] Verifying if .mailmap-exclude exists'
+    test -e .mailmap-exclude \
+      ||  {
+            echo '  [ERROR] Please add a .mailmap-exclude file in the root of the repo'
+            exit 1
+          }
```
Comment (kamadorueda) [inline default.nix]: I see the value in having an exclusion list, especially for bot accounts, but I don't think having an exclusion list should be mandatory as this implies. Most projects will do fine without excludes, others will prefer explicitly mapping the bot accounts. I'm not opposed to picking the file if it exists though, I just don't think it should be mandatory

If the excludes are not too many, we may also accept a command line flag `--exclude '...'`

File: default.nix [hunk 2]
```diff
@@ -1,100 +1,121 @@
...
-        line_number=$((line_number + 1))
-        if ! echo "$mapping" | grep -qP "$format"
+        if [[ $author =~ $exclude ]]
         then
-          echo "  [ERROR] Expected: $format"
-          echo "          Line $line_number: $mapping"
-          exit 1
+          echo "  [INFO] Excluded: $author"
+          found=true
+          break
```
Comment (kamadorueda) [inline default.nix]: would a `continue 2` work here?

File: default.nix (hunk 1 above)
Comment (rohaquinlop) [reply default.nix]: Done!

File: default.nix (hunk 2 above)
Comment (rohaquinlop) [reply default.nix]: Done!

**Review by kamadorueda — APPROVED** [review]

**Pull request merged**